Bottom: 09e29ea5aa798c9b53ccdc3cb351c60818e29c17
Top:    29dc8d3a9a39c4d50895b3f70d2471d009bcfe7f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-04-05 22:30:18 +0100

Add profiling to file load

















---

diff --git a/src/file.c b/src/file.c
index b6c67b9..018b272 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
